contrib/masterfiles-stage/masterfiles-stage.sh

```diff
@@ -1,45 +1,43 @@
 #!/usr/bin/env bash
-# Authors: Ted Zlatanov <tzz@lifelogs.com>, Nick Anderson <nick@cmdln.org>, Mike Weilgart
-DIRNAME=$(dirname $0);
+# Authors: Ted Zlatanov <tzz@lifelogs.com>, Nick Anderson <nick@cmdln.org>, Mike Weilgart <mikeweilgart@gmail.com>
+mydirname="$(dirname "$0")";
 
 # Load option parsing
-source "${DIRNAME}/options.sh"
+# Sets vars: PARAMS, MASTERDIR, verbose_mode, debug_mode
+source "${mydirname}/options.sh" || { echo "options.sh couldn't be sourced" >&2 ; exit 17 ; }
 
-# Load common functionaly, upstream implementations
-source "${DIRNAME}/common.sh"
+# Load common functionality, upstream implementations
+source "${mydirname}/common.sh" || { echo "common.sh couldn't be sourced" >&2 ; exit 17 ; }
 
-MASTERDIR=$opt_deploy_dir
-PARAMS=$opt_params_file
+[ -f "$PARAMS" ] || error_exit "ERROR: Missing '$PARAMS'"
 
-# If PARAMS is unset, then we look to the default location
-#PARAMS=${PARAMS:-/opt/cfengine/dc-scripts/params.sh}
-if [ ! -f "$PARAMS" ]; then
-  error_exit "ERROR: Missing '$PARAMS'"
-fi
+source "$PARAMS"
 
-source $PARAMS
-
-  # We probably want a different temporary location for each remote repository
-  # so that we can avoid conflicts and potential confusion.
-  # Example: 
-  # ROOT="/opt/cfengine/masterfiles_staging"
-  # PARAMS="/var/cfengine/policychannel/production_1.sh"
-  # TMP=/opt/cfengine/masterfiles/staging/_tmp_var_cfengine_policychannel_production_1_sh
-
-  TRANSLATED_PARAMS=$(echo $PARAMS | tr [./] _)
-  STAGING_DIR="${ROOT}/_tmp${TRANSLATED_PARAMS}"
```

Comment:
The version that ships with enterprise currently has:

```
[root@hub dc-scripts]# grep ROOT masterfiles-stage.sh 
  TMP="${ROOT}_tmp"
```

Now I made a similar modification to your proposed changes and I still have some breakage in Design Center that I am trying to track down. With the script we ship everything works fine, but I need to make sure the new and improved version will continue to work. It really seems like it should be as easy using paths that are next to ROOT instead of inside of it. So I am still confused as to why with that modification it still breaks design center.